### PR TITLE
Tidy Up

### PR DESCRIPTION
**Pull request by joshuatcasey**
## Summary
Tidy up, mostly godoc, README, and test cases

## Use Cases
This is mostly so that the diff for an upcoming live reload PR is much smaller and more narrowly scoped to live reload

### CODE REVIEW

File: detect.go [hunk 1]
```diff
@@ -23,18 +24,21 @@ type BuildPlanMetadata struct {
...
-// will provide and require `nginx-start`. This is unusual, but will allow the
-// buildpack Build function access to which web server start command is needed,
-// since the requirements are not easily checked otherwise.
+// This buildpack has two requirement groups:
+// One for HTTPD in which the following are required at launch time:
+// - "php"
+// - "php-fpm"
+// - "httpd"
+// - "httpd-config"
+// Another for HTTPD in which the following are required at launch time:
```
Comment (sophiewigmore) [inline detect.go:33]:
```suggestion
// Another for Nginx in which the following are required at launch time:
```

File: detect.go [hunk 2]
```diff
@@ -23,18 +24,21 @@ type BuildPlanMetadata struct {
...
-// One for HTTPD in which "php", "php-fpm", "httpd"
-// and "httpd-config" are required at launch-time.
-// The second one for for Nginx in which "php", "php-fpm", "nginx"
-// and "nginx-config" are required at launch-time.
-
-// This buildpack will always detect, and in the case of HTTPD, the buildpack
-// will provide and require `httpd-start`. In the case of Nginx, the buildpack
-// will provide and require `nginx-start`. This is unusual, but will allow the
-// buildpack Build function access to which web server start command is needed,
-// since the requirements are not easily checked otherwise.
```
Comment (sophiewigmore) [inline detect.go:37]: Why is all of this context removed?

File: detect.go [hunk 3]
```diff
@@ -108,26 +112,10 @@ func Detect() packit.DetectFunc {
...
 		nginxFpmPlan.Requires = append(baseRequirements, nginxFpmPlan.Requires...)
-
-		plans := []packit.BuildPlan{httpdFpmPlan, nginxFpmPlan}
+		httpdFpmPlan.Or = []packit.BuildPlan{nginxFpmPlan}
 
 		return packit.DetectResult{
-			Plan: or(plans...),
+			Plan: httpdFpmPlan,
 		}, nil
 	}
```
Comment (sophiewigmore) [inline detect.go:120]: this change is more confusing to me than the code we had before.

File: detect.go (hunk 2 above)
Comment (joshuatcasey) [reply detect.go:37]: The buildpack no longer has any reference to `httpd-start` or `nginx-start`, so the entire block was simply incorrect.

File: detect.go (hunk 1 above)
Comment (joshuatcasey) [reply detect.go]: fixed

File: detect.go (hunk 3 above)
Comment (joshuatcasey) [reply detect.go:120]: Fair. This is a shortcut to achieve the same result. The tests clearly require the structure where the `httpdFpmPlan` is the root plan with an `Or: nginxFpmPlan`.

I suspect there's some context on why the spec is the way it is due to backwards compatibility, but I really don't know. But the code as-written delivers the expected plan structure and it's somewhat simpler than before so I think it's useful.

Comment (joshuatcasey) [reply detect.go:120]: @sophiewigmore does this block merging?

Comment (sophiewigmore) [reply detect.go:120]: Yea, I'd prefer to put it back to the way it was before, which to me was more readable.

Comment (joshuatcasey) [reply detect.go]: @sophiewigmore look now?

**Review by sophiewigmore — APPROVED** [review]

**Pull request merged**